pgmpy/factors/CPD.py

```diff
@@ -327,19 +327,15 @@ def marginalize(self, variables, inplace=True):
         >>> cpd_table = TabularCPD('grade', 2,
         ...                        [[0.7, 0.6, 0.6, 0.2],[0.3, 0.4, 0.4, 0.8]],
         ...                        ['intel', 'diff'], [2, 2])
-        >>> cpd_table.marginalize('diff')
+        >>> cpd_table.marginalize(['diff'])
         >>> cpd_table.get_cpd()
-        array([[ 0.48484848,  0.4       ],
-               [ 0.51515152,  0.6       ]])
+        array([[ 0.65,  0.4 ],
+                [ 0.35,  0.6 ]])
         """
-        if inplace:
-            tabular_cpd = self
-        else:
-            evidence = self.variables[1:]
-            evidence_card = self.cardinality[1:]
-            tabular_cpd = TabularCPD(self.variable, self.variable_card,
-                                     self.get_cpd(), evidence,
-                                     evidence_card)
+        if inplace and self.variable in variables:
```

Comment:
@anaviltripathi I think we shouldn't allow it at all, no matter if it's inplace or not. What do you think?